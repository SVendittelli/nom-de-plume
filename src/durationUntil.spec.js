const { DateTime } = require('luxon');

const durationUntil = require('./durationUntil');

describe('durationUntil', () => {
  test('same day', () => {
    const start = DateTime.local(2020, 1, 1);
    const end = DateTime.local(2020, 1, 1);

    const { duration, humanReadableDuration } = durationUntil(start, end);
    expect(duration == 0).toBeTruthy();
    expect(humanReadableDuration).toMatchSnapshot();
  });

  test('day in the future', () => {
    const start = DateTime.local(2020, 1, 1);
    const end = DateTime.local(2020, 2, 1);

    const { duration, humanReadableDuration } = durationUntil(start, end);
    expect(duration > 0).toBeTruthy();
    expect(humanReadableDuration).toMatchSnapshot();
  });

  test('day in the past', () => {
    const start = DateTime.local(2020, 1, 1);
    const end = DateTime.local(2019, 1, 1);

    const { duration, humanReadableDuration } = durationUntil(start, end);
    expect(duration < 0).toBeTruthy();
    expect(humanReadableDuration).toMatchSnapshot();
  });
});
