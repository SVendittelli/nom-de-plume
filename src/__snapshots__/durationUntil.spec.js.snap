// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`durationUntil day in the future 1`] = `"1 month, 1 day"`;

exports[`durationUntil day in the past 1`] = `"1 year"`;

exports[`durationUntil same day 1`] = `""`;
